assets/js/base/utils/legacy-events.ts

```diff
@@ -50,6 +59,23 @@ export const triggerFragmentRefresh = (): void => {
 	}, 50 );
 };
 
+export const triggerAddingToCartEvent = (): void => {
+	dispatchEvent( 'wc-blocks_adding_to_cart', {
+		bubbles: true,
+		cancelable: true,
+	} );
+};
+
+export const triggerAddedToCartEvent = ( {
+	invalidateCartData,
```

Comment:
Is it worth giving this a default value of `false`? Given `undefined` is a falsey value anyway and we can change its usage in this file from:

`yield triggerAddedToCartEvent( { invalidateCartData: false } );`

to 

`yield triggerAddedToCartEvent();`

Reply:
> Is it worth giving this a default value of `false`

Good point. I took another look at the code and I think the way I wrote it was a bit confusing, because it's true that here `invalidateCartData` defaults to false, but then in the React components I wrote this check: `! eventDetail || eventDetail.invalidateCartData !== false`, which would default to true if it's undefined.

In 17747aae5ab53091b9d2438df8386ca5fe9e9a98, I refactored the code so now the property is named `preserveCartData` instead of `invalidateCartData`. This way it can default to false in all cases, I think this makes the code easier to understand, but maybe I'm just overcomplicating this? :sweat_smile: 

> define a type for it

The type comes from `AddToCartEventDetail`, right?

Reply:
>I refactored the code so now the property is named preserveCartData instead of invalidateCartData. This way it can default to false in all cases, I think this makes the code easier to understand, but maybe I'm just overcomplicating this?

Thanks! I think this makes more sense and its easier to understand!

>The type comes from AddToCartEventDetail, right?

You're absolutely right, my bad sorry!

All LGTM 🚀 😄